python/ray/tune/api_v2/test_tuner.py

```diff
@@ -0,0 +1,700 @@
+                processed_datasets = this_datasets
+
+            run_config = self_run_config or {}
+            if override_run_config:
+                run_config.update(override_run_config)
+
+            scaling_config = self_scaling_config or {}
+            if override_scaling_config:
+                scaling_config.update(override_scaling_config)
+
+            def update(name: str, arg: Any, config: Dict[str, Any]):
+                # Pop config items to allow fo deep override
+                config_item = config.pop(name, None)
+                if isinstance(arg, dict) and isinstance(config_item, dict):
+                    arg.update(config_item)
+                    return arg
+                return config_item or arg
+
+            updated_kwargs = {
+                name: update(name, arg, config) for name, arg in self_kwargs.items()
+            }
+
+            # Update with remaining config ite s
+            updated_kwargs.update(config)
+
+            if checkpoint_dir:
+                checkpoint = LocalStorageCheckpoint(path=checkpoint_dir)
+            else:
+                checkpoint = self_resume_from_checkpoint
+
+            self.train_fn(
+                run_config=run_config,
+                scaling_config=scaling_config,
+                datasets=processed_datasets,
+                checkpoint_dir=checkpoint_dir,
+                **updated_kwargs,
+            )
+
+        trainable = wrap_function(internal_train_fn)
+        trainable.__name__ = self.train_fn.__name__
+
+        # Monkey patching the resource requests for dynamic resource
+        # allocation
+        def resource_request(config):
+            config = copy.deepcopy(config)
+
+            scaling_config = {
+                "num_actors": 0,
+                "cpus_per_actor": 1,
+                "gpus_per_actor": 1,
+                "resources_per_actor": None,
+            }
+            if self_scaling_config:
+                scaling_config.update(self_scaling_config)
+
+            override_scaling_config = config.pop("scaling_config", None)
+            if override_scaling_config:
+                scaling_config.update(override_scaling_config)
+
+            return self.resource_fn(scaling_config)
+
+        trainable.default_resource_request = resource_request
+
+        def postprocess_checkpoint(config: Dict[str, Any], checkpoint: Checkpoint):
+            checkpoint.preprocessor = config.get("preprocessor", None)
+
+        trainable.postprocess_checkpoint = postprocess_checkpoint
+
+        return trainable
+
+
+class XGBoostTrainer(FunctionTrainer):
+    def train_fn(
+        self,
+        run_config: RunConfig,
+        scaling_config: ScalingConfig,
+        datasets: Dict[str, ray.data.Dataset],
+        checkpoint_dir: Optional[str],
+        label: str,
+        params: Optional[Dict[str, Any]],
+        **kwargs,
+    ):
+        # train_dataset = datasets["train_dataset"]
+        train_dataset = gen_dataset_func()
```

Comment:
This is generating a new dataset on each training run, but shouldn't the training dataset be pulled from the Tune-provided param space?

Reply:
ah, you are right. I forgot to change this back after testing something else.